### ISSUE TITLE
typing.NamedTuple self considered non-sequence in methods

### ISSUE THREAD

**Issue by skeeto** [PyCQA/pylint#5312]
### Bug description

When inheriting from `typing.NamedTuple`, the `self` value in methods is not recognized as a sequence. Unpacking `self` results in E0633 (unpacking-non-sequence). This only happens in its own methods, and there is no issue with unpacking elsewhere. Example:

```py
# pylint: disable=C0103,C0114,C0115,C0116
import typing

class V2(typing.NamedTuple):
    x: float
    y: float

    def sum(self):
        x, y = self  # <- pylint complains about unpacking
        return x + y
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
example.py:9:8: E0633: Attempting to unpack a non-sequence defined at line 4 (unpacking-non-sequence)
```

### Expected behavior

No complaint since unpacking `self` is valid here.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.5
Python 3.10.0 (default, Oct  5 2021, 10:05:34) [GCC 11.2.0]
```
